### ISSUE TITLE
Make NuxtLink scroll to anchor by default

### ISSUE THREAD

**Issue by iBobik** [nuxt/framework#8424]
### Describe the feature

```
<NuxtLink to="/#shop">Buy now</NuxtLink>
```

Currently we have to use [inconvenient workaround](https://github.com/nuxt/framework/issues/5185#issuecomment-1140446513), but it should by default scroll to anchor.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by Lyokolux** [comment]
As Nuxt ist MPA first, this would make sense. If nuxt is an SPA, then the anchor should allow to focus the item id IMHO.

**Comment by iBobik** [comment]
My issue is not about SPA vs MPA, but normal website with presentation and e-shop. On presentation pages it is common to link to specific part of the page. Global navigation has links like /about, /terms, / (homepage), /#product (product section on the homepage).

NuxtLink seems designed to replace all kind of links, so we does not have to think if to use this or `<a href`, so I expected it also handles "scroll to" feature of the anchor.

**Comment by ChrisScrumping** [comment]
I agree with iBobik, in fact in your example currently if you are on `/about `and link to `/#product` it just loads the home page and does not scroll to the anchor point even with the work around - unless I am missing something?

The solution is to use `external="true` but this breaks transitions?

**Comment by ChrisScrumping** [comment]
I got this working as per my comment here: https://github.com/nuxt/framework/discussions/5561#discussioncomment-4257990
